source/exe/main_common.cc

```diff
@@ -0,0 +1,69 @@
+#include <iostream>
+#include <memory>
+
+#include "common/common/compiler_requirements.h"
+#include "common/event/libevent.h"
+#include "common/local_info/local_info_impl.h"
+#include "common/network/utility.h"
+#include "common/stats/stats_impl.h"
+#include "common/stats/thread_local_store.h"
+
+#include "exe/hot_restart.h"
+
+#include "server/config_validation/server.h"
+#include "server/drain_manager_impl.h"
+#include "server/options_impl.h"
+#include "server/server.h"
+#include "server/test_hooks.h"
+
+#include "ares.h"
+#include "spdlog/spdlog.h"
+
+namespace Envoy {
+namespace Server {
+
+class ProdComponentFactory : public ComponentFactory {
+public:
+  // Server::DrainManagerFactory
+  DrainManagerPtr createDrainManager(Instance& server) override {
+    return DrainManagerPtr{new DrainManagerImpl(server)};
+  }
+
+  Runtime::LoaderPtr createRuntime(Server::Instance& server,
+                                   Server::Configuration::Initial& config) override {
+    return Server::InstanceUtil::createRuntime(server, config);
+  }
+};
+
+} // Server
+
+int main_common(Envoy::OptionsImpl& options, Envoy::Server::HotRestartImpl& restarter) {
+  Envoy::Event::Libevent::Global::initialize();
```

Comment:
@dnoe nit: Since everything in this function is now in Envoy namespace, it would be nice to not prefix all things with Envoy:: just to make code less verbose. NBD but if you feel like a follow up.